src/FieldWarning/Assets/Ingame/UI/InputManager.cs

```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
      */
     public class InputManager : MonoBehaviour
     {
-        private Texture2D _firePosReticle;
-        private Texture2D _primedReticle;
+        private readonly List<SpawnPointBehaviour> _spawnPointList = new List<SpawnPointBehaviour>();
 
-        private List<SpawnPointBehaviour> _spawnPointList = new List<SpawnPointBehaviour>();
-        private ClickManager _rightClickManager;
-
-        public enum MouseMode { normal, purchasing, firePos, reverseMove, fastMove };
-        public MouseMode CurMouseMode { get; private set; } = MouseMode.normal;
+        private Vector3 _boxSelectStart;
 
         private BuyTransaction _currentBuyTransaction;
 
-        private MatchSession _session;
-        public MatchSession Session {
-            get {
-                return _session;
-            }
+        private Texture2D _firePosReticle;
 
-            set {
-                if (_session == null)
-                    _session = value;
-            }
-        }
+        private Dictionary<string, Func<Commands, object>> _hotkeyActions;
 
-        private SelectionManager _selectionManager;
+        private Texture2D _primedReticle;
 
-        private PlayerData _localPlayer {
-            get {
-                return Session.LocalPlayer.Data;
-            }
-        }
+        private ClickManager _rightClickManager;
 
-        private Vector3 _boxSelectStart;
+        private SelectionManager _selectionManager;
 
-        void Awake()
-        {
-            _selectionManager = new SelectionManager();
-            _selectionManager.Awake();
-        }
+        private MatchSession _session;
 
-        void Start()
+        public enum Events
         {
-            _firePosReticle = (Texture2D)Resources.Load("FirePosTestTexture");
-            if (_firePosReticle == null)
-                throw new Exception("No fire pos reticle specified!");
-
-            _primedReticle = (Texture2D)Resources.Load("PrimedCursor");
-            if (_primedReticle == null)
-                throw new Exception("No primed reticle specified!");
-
-            _rightClickManager = new ClickManager(1, MoveGhostsToMouse, OnOrderShortClick, OnOrderLongClick, OnOrderHold);
+            Clicked,
+            Down,
+            Released
         }
 
-        void Update()
+        public enum Modes
```

Comment:
Plural in an enum seems unnatural to me and SO agrees:
https://stackoverflow.com/questions/15755955/naming-of-enums-in-java-singular-or-plural
https://stackoverflow.com/questions/1405851/enum-naming-convention-plural

I would also really prefer a more expressive name. What modes are these? Firing modes? Launch modes? I don't understand why we're trying to save 5 characters here.